docs/articles/documentation/guides/continuous-integration/github-actions-and-browserstack.md

```diff
@@ -0,0 +1,252 @@
+---
+layout: docs
+title: Run Tests on BrowserStack with GitHub Actions
+permalink: /documentation/guides/continuous-integration/github-actions-and-browserstack.html
+redirect_from:
+  - /documentation/recipes/integrating-testcafe-with-ci-systems/github-actions-and-browserstack.html
+  - /documentation/continuous-integration/github-actions-and-browserstack.html
+---
+# Run Tests on BrowserStack with GitHub Actions
+
+This topic describes use of [Run TestCafe action](https://github.com/DevExpress/testcafe-action) to integrate TestCafe into the [GitHub Actions](https://docs.github.com/en/actions/automating-your-workflow-with-github-actions) build process. Tests are performed in [BrowserStack](https://automate.browserstack.com/) cloud testing service.
+
+* [Step 1 - Create a Workflow](#step-1---create-a-workflow)
+* [Step 2 - Create a Job](#step-2---create-a-job)
+* [Step 3 - Provide your BrowserStack credentials](#step-3---provide-browserstack-credentials)
+* [Step 4 - Add a Step That Fetches The Repository](#step-4---add-a-step-that-fetches-the-repository)
+* [Step 5 - Add a Step To Install TestCafe BrowserStack plugin](#step-5---add-a-step-to-install-testcafe-browserstack-plugin)
+* [Step 6 - Add a Step That Runs TestCafe](#step-6---add-a-step-that-runs-testcafe)
+* [Action Options](#action-options)
+  * [args](#args)
+  * [version](#version)
+* [Example](#example)
+
+## Step 1 - Create a Workflow
+
+Create a YAML file (for instance, `testcafe-workflow.yml`) in the `.github/workflows` directory in your repository.
+
+Specify the [workflow name](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#name) and the [event](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#on) that triggers this workflow.
+
+```yml
+name: End-to-End Tests
+on: [push]
+```
+
+In this example, the workflow runs when you push changes to the repository.
+
+## Step 2 - Create a Job
+
+Create a job that runs the TestCafe tests.
+
+Provide the [job name](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idname) and specify the [type of machine](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on) that should run the job.
+
+```yml
+name: End-to-End Tests
+on: [push]
+
+jobs:
+  test:
+    name: Run TestCafe Tests
+    runs-on: windows-latest
+```
+
+This job runs on a GitHub-hosted virtual machine with the latest Windows version. `test` is the [job ID](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_id) that must be unique to the `jobs` object.
+
+## Step 3 - Provide BrowserStack Credentials
+
+In order for TestCafe to be able to use browsers, provided by BrowserStack, valid Username and Access Key are necessary. Values are unique to your BrowserStack account and can be obtained from your [account settings](https://www.browserstack.com/accounts/settings).
```

Comment:
"In order for TestCafe to be able to use browsers, provided by BrowserStack, valid Username and Access Key are necessary."

Does BrowserStack provide browsers? Or does it provide the Username and Access Key?

Reply:
Both. BrowserStack is a service that hosts (i.e. provides) browsers for users. 

To access the service, user need to use Username and Access Key (essentially, login and password). User logs onto BrowserStack website in their browser, copies Username and Access Key from there, punches it into TestCafe (step 3) so it can call BrowserStack on their (user's) behalf.

Therefore, user provides BrowserStack's creds **to** TestCafe

Am I being clear?

Reply:
Thank you! Please see my suggestion below.

Reply:
```suggestion
In order for TestCafe to use browsers provided by BrowserStack, valid BrowserStack credentials (Username and Access Key) are required. These credentials are unique to your BrowserStack account and can be obtained from your [account settings](https://www.browserstack.com/accounts/settings) page.
```